### PR TITLE
factor out creation of default encoder into a method

### PR DESCRIPTION
**Pull request by tobyink**
Reduces a bit of repetition and avoids loading JSON::PP or Cpanel::JSON::XS unless they're needed.

(In the current release, even if I want to use my custom JSON encoder, `Cpanel::JSON::XS` or `JSON::PP` will always be loaded at compile time.)

### CODE REVIEW

File: lib/JSON/Slurper.pm [hunk 1]
```diff
@@ -15,10 +15,22 @@ our %EXPORT_TAGS = (
     spurt_auto => [qw(-auto_ext spurt_json)],
 );
 
-use constant JSON_XS => $ENV{JSON_SLURPER_NO_JSON_XS}                                          ? do { require JSON::PP; undef }
-                     : eval { require Cpanel::JSON::XS; Cpanel::JSON::XS->VERSION('4.09'); 1 } ? 1
-                     : do { require JSON::PP; undef };
 my $DEFAULT_ENCODER;
+sub _build_default_encoder {
```
Comment (srchulo) [inline lib/JSON/Slurper.pm:19]: Thanks! I should've done this from the beginning :)

File: lib/JSON/Slurper.pm [hunk 2]
```diff
@@ -27,11 +39,7 @@ sub new {
     if (exists $args{encoder}) {
         $encoder = _validate_encoder(delete $args{encoder});
     } else {
-        $encoder = $DEFAULT_ENCODER ||=
-          JSON_XS
-          ? Cpanel::JSON::XS->new->utf8->pretty->canonical->allow_nonref->allow_blessed->convert_blessed->escape_slash
-          ->stringify_infnan
-          : JSON::PP->new->utf8->pretty->canonical->allow_nonref->allow_blessed->convert_blessed->escape_slash;
+        $encoder = ($DEFAULT_ENCODER ||= $class->_build_default_encoder);
```
Comment (srchulo) [inline lib/JSON/Slurper.pm:42]: Why `$class->_build_default_encoder` instead of just `_build_default_encoder()`?

**Comment by tobyink** [comment]
Allows JSON::Slurper to be easily subclassed with a different default.

**Comment by srchulo** [comment]
Awesome, thanks! :)

**Pull request merged**